### DISCUSSION TITLE
custom types and enums in slint rust

### DISCUSSION THREAD

**Discussion by disoreks** [slint-ui/slint#4736] [Q&A]
hey, 

i was wondering why the types supported by models are limited to f32/i32 and what's the correct thing to do if i'm working with bigger numbers in my data like f64/u64+

and how do i define enums variants that store values in the .slint file?
like:
```
enum Quantity {
    Single(i32),
    Grouped(i32),
}
```
thanks :)

**Comment by tronical** [comment]
I think `f64` and `u64` would make sense to add as data types in the language.

Support for enums that store values is not implemented yet. It's a bit unfortunate as while it is native to Rust as target language, it's not native in C++ (it's a little awkward requiring wrappers). But it could be done, and it would be nice to have :)

**Comment by ogoffart** [comment]
I think this is a valuable feature.  This has also been discussed in https://github.com/slint-ui/slint/issues/1307
